### PR TITLE
Integrations Updates

### PR DESCRIPTION
**Pull request by ben-chain**
## Description
This PR updates a few small things which came up as a result of more extensive integrations testing, particularly:
- the mock ECDSA account was not bumping its nonce properly.
- the safety checker bytes have been updated -- needed to reflect in the safety checker.
## Questions
N/A

### Fixes
- Fixes # 17

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol [hunk 1]
```diff
@@ -78,7 +78,7 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
 
             // EVM doesn't tell us whether a contract creation failed, even if it reverted during
             // initialization. Always return `true` for our success value here.
```
Comment (K-Ho) [inline contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol:80]: update comment -> "return true if created address exists:

File: contracts/optimistic-ethereum/mockOVM/accounts/mockOVM_ECDSAContractAccount.sol [hunk 2]
```diff
@@ -58,14 +58,23 @@ contract mockOVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
 
             // EVM doesn't tell us whether a contract creation failed, even if it reverted during
             // initialization. Always return `true` for our success value here.
-            return (true, abi.encode(created));
+            return (created != address(0), abi.encode(created));
```
Comment (K-Ho) [inline contracts/optimistic-ethereum/mockOVM/accounts/mockOVM_ECDSAContractAccount.sol:61]: I think we can get rid of this file with the EOA updates ( not sure what we needed this mock for in the first place)

**Review by K-Ho — APPROVED** [review]
lgtm!!!

File: contracts/optimistic-ethereum/mockOVM/accounts/mockOVM_ECDSAContractAccount.sol (hunk 2 above)
Comment (ben-chain) [reply contracts/optimistic-ethereum/mockOVM/accounts/mockOVM_ECDSAContractAccount.sol:60]: this is actually exclusively used in the JSVM, leaving for those purposes.

**Pull request merged**